### ISSUE TITLE
Configuration failure when using Register-PSRepository on PS Core but not PS

### ISSUE THREAD

**Issue by Creetor** [PowerShell/PowerShell#9611]
Hello.  I have been automating tasks involving Azure DevOps' NuGet Feed with PowerShell Core on Windows 10 and have found a very odd inconsistency.  In short, if I try to register the NuGet Feed as a PS Repository through PS Core, I receive a warning and any subsequent attempts to access the PS Repo fail (before and after a console restart). The **weird part** is that if I use the same code in PowerShell to register the PS Repo and then open a new PS Core console, the follow-up commands work as expected.

# Steps to reproduce
1. **May or May Not Matter** - Follow the steps from [Use Azure Artifacts as a private PowerShell repository](https://docs.microsoft.com/en-us/azure/devops/artifacts/tutorials/private-powershell-library?view=azure-devops&viewFallbackFrom=vsts) until "Connecting to the feed as a PowerShell repo".
2. Attempt to add the NuGet Feed as a PS Repository and then use it:

```powershell
Register-PSRepository -Name "PowershellAzureDevopsServices" -SourceLocation "https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/" -PublishLocation "https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/" -InstallationPolicy Trusted
$password = ConvertTo-SecureString 'REDACTED' -AsPlainText -Force
$credsAzureDevopsServices = New-Object System.Management.Automation.PSCredential 'REDACTED', $password
Find-Module -Repository PowershellAzureDevopsServices -Credential $credsAzureDevopsServices
```

# Expected behavior
```PowerShell
Register-PSRepository -Name "PowershellAzureDevopsServices" -SourceLocation "https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/" -PublishLocation "https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/" -InstallationPolicy Trusted
$password = ConvertTo-SecureString 'REDACTED' -AsPlainText -Force
$credsAzureDevopsServices = New-Object System.Management.Automation.PSCredential 'REDACTED', $password
Find-Module -Repository PowershellAzureDevopsServices -Credential $credsAzureDevopsServices

Version              Name                                Repository           Description
-------              ----                                ----------           -----------
0.0.1                Get-Hello                           PowershellAzureDevo… Say Hello PowerShellGet
```

* This is also how PS responds and after I add Register the Repo in PS, the other command works in a new instance of PS Core.

# Actual behavior

```none
Register-PSRepository -Name "PowershellAzureDevopsServices" -SourceLocation "https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/" -PublishLocation "https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/" -InstallationPolicy Trusted

WARNING: Unable to resolve pack
age source 'https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/'.

$password = ConvertTo-SecureString 'REDACTED' -AsPlainText -Force
$credsAzureDevopsServices = New-Object System.Management.Automation.PSCredential 'REDACTED', $password
Find-Module -Repository PowershellAzureDevopsServices -Credential $credsAzureDevopsServices

WARNING: Unable to resolve package source 'https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/'.
```

# Environment data

## PowerShell Core

```PowerShell
PS C:\dev> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## PowerShell

``` PowerShell
PS C:\WINDOWS\system32> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.17134.590
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.590
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by rlvandaveer** [comment]
Interestingly enough, I followed the same tutorial today and had a similar issue. Though in my case, my results have been intermittently successful from PowerShell Core on a Mac, (worked the first time, didn't work any subsequent times until moments ago), and completely unsuccessful from Windows Powershell 5.1 on a Windows Server 2016 box. Registering the repository and trying to install a module results in the same message: `WARNING: Unable to resolve package source 'https://REDACTED.pkgs.visualstudio.com/_packaging/REDACTED/nuget/v2/'`. I was able to publish my package and can see the package using the nuget CLI on both systems, but that is using the V3 URL.

**Comment by iSazonov** [comment]
I think the issue should be moved to PowerShellGet repo https://github.com/PowerShell/PowerShellGet

**Comment by rlvandaveer** [comment]
I would tend to agree that this is an issue with PowerShellGet and/or possibly PackageManagement.

I believe I have figured out why my experience was seemingly intermittent on PowerShell Core and just didn't work on Windows Powershell. I had added a nuget source for my repository using the V3 URL following the tutorial referenced above and it was named identically to my PSRepo. I don't understand why, but operations for that PSRepo always failed with the unable to resolve package source warning. When I changed my nuget source to use the V2 URL and restarted my PowerShell session, it worked. My experience in Core was seemingly intermittent because I had not restarted my PowerShell session after changing my nuget source URL. Since it didn't appear to work in Core I never tried it in Windows PowerShell.

I've swapped out the URL for my nuget source in PowerShell Core and Windows PowerShell 5.1, restarted my sessions and I am no longer seeing the issue.
